core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -993,7 +993,7 @@ public static SqlDataTypeSpec convertTypeToSpec(RelDataType type,
     assert typeName != null;
 
     final SqlTypeNameSpec typeNameSpec;
-    if (isAtomic(type)) {
+    if (isAtomic(type) || isNull(type)) {
```

Comment:
This change seems not necessary any more ?

Reply:
I would like to keep it as far as it fixes and issue in a public utility method. But thank you for spotting that SQL tests no longer hit it, so I covered this fix in SqlTypeUtilTest.